libscript/src/module-java.cpp

```diff
@@ -0,0 +1,105 @@
+/* Copyright (C) 2003-2015 LiveCode Ltd.
+ 
+ This file is part of LiveCode.
+ 
+ LiveCode is free software; you can redistribute it and/or modify it under
+ the terms of the GNU General Public License v3 as published by the Free
+ Software Foundation.
+ 
+ LiveCode is distributed in the hope that it will be useful, but WITHOUT ANY
+ WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ for more details.
+ 
+ You should have received a copy of the GNU General Public License
+ along with LiveCode.  If not see <http://www.gnu.org/licenses/>.  */
+
+#include <foundation.h>
+#include <foundation-auto.h>
+
+static bool TryToInitializeJava()
+{
+	if (!MCJavaInitialize())
+		return MCErrorCreateAndThrow(kMCGenericErrorTypeInfo, "reason",
+                                     MCSTR("could not initialize java"),
+                                     nullptr);
+
+	return true;
+}
+
+extern "C" MC_DLLEXPORT_DEF MCTypeInfoRef MCJavaObjectTypeInfo() { return MCJavaGetObjectTypeInfo(); }
+
+////////////////////////////////////////////////////////////////////////////////////////////////////
+
+extern "C" MC_DLLEXPORT_DEF void MCJavaStringFromJString(MCJavaObjectRef p_object, MCStringRef &r_string)
+{
+	if (!TryToInitializeJava())
+		return;
+	
+    if (!MCJavaConvertJStringToStringRef(p_object, r_string))
+        MCErrorCreateAndThrow(kMCGenericErrorTypeInfo, "reason",
+                              MCSTR("couldn't convert java object to string"),
+                              nullptr);
+}
```

Comment:
I feel like it would be good to have distinct type infos for the the Java-specific errors rather than using the generic error type.